akka-http/src/main/scala/akka/http/scaladsl/unmarshalling/Unmarshaller.scala

```diff
@@ -154,7 +148,40 @@ object Unmarshaller
         s"Right failure: ${right.getMessage}, " +
         s"Left failure: ${left.getMessage}")
 
+  /**
+   * Signals that unmarshalling failed because the entity content-type did not match one of the supported ranges.
+   * This error cannot be thrown by custom code, you need to use the `forContentTypes` modifier on a base
+   * [[akka.http.scaladsl.unmarshalling.Unmarshaller]] instead.
+   */
+  final case class UnsupportedContentTypeException(
+    supported:   Set[ContentTypeRange],
+    contentType: Option[ContentType]   = None)
```

Comment:
Is the only reason to have this as an option binary compatibility? That's kind of unfortunate... perhaps we should consider introducing a singleton `ContentType` to use in that case?

Reply:
no, the reason is `Content-Type` is not always available

Reply:
Using an default value is a bad ideas as with those it becomes even harder to keep any bin compat as one later loses the capability to do overloads properly. Please remove the default value, we need to provide methods that achieve the same effect, which I think have been added.

Reply:
Perhaps we could make the naming more descriptive here? `actual`?

Reply:
@ktoso removed the default value